### DISCUSSION TITLE
Missing Datatypes

### DISCUSSION THREAD

**Discussion by magr2** [strictdoc-project/reqif#63] [Q&A] [answered]
Hello,
I recently started working with ReqIf and and came across this library.  However I have the following problem: when I try to load my ReqIF into Python, there is a NotImplementdError, because the datatypes Real and Date, which are part of the newest specification, but are not yet implemented. My question is: what would I need to do to add them ? Kind regards magr

**Comment by stanislaw** [comment]
Hi @magr2,

Thank you for your interest in using this library. I see two options:

1) You send me a [minimal reproducer](https://en.wikipedia.org/wiki/Minimal_reproducible_example) and I will quickly implement it.

2) You do it yourself and send a Pull Request to this repository with a fix. For this, you will have to add a corresponding integration test (find the right folder and follow the examples here: https://github.com/strictdoc-project/reqif/tree/main/tests/integration). You can run the tests using `invoke test-integration` and check `invoke --list` in general when you are in the reqif project folder.

I am happy to proceed with either option!

**Reply by magr2** [reply]
Hello @stanislaw ,

Thank you for your quick answer. I guess I will try to find a fix myself and get to know the library a bit better in the process.

**Comment by magr2** [comment]
Hi @stanislaw,

while trying to test my implementation of the data types, I encountered an issue:
When trying to run "invoke test-integration", I get the error
find tests -type f \( -name '*.script' \) -or -type d \( -name '*.dSYM' -or -name 'sandbox' -or -name 'Output' -or -name 'output' \) -not -path "**Expected**" -not -path "**Input**"
File not found - '*.script'
even without adding any test. Do I have to make this script-file on my own ?

**Reply by stanislaw** [reply]
Which system are you running this on?

**Reply by magr2** [reply]
I am running this on Windows 11

**Reply by magr2** [reply]
@stanislaw do I have to run this on a Linux system ?

**Reply by stanislaw** [reply]
Yes, the Windows CI is supported by ReqIF but these tests are run using Bash: https://github.com/strictdoc-project/reqif/blob/5537d81bab1fbaf72bba91d4193fc52df7c3eb71/.github/workflows/ci-windows.yml#L46. The `find` command has not been made portable yet.

**Reply by magr2** [reply]
@stanislaw sorry for the delay. I finally had the opportunity to test the code I wrote. However during the test I got several errors, which dont seem to be related to my implementation. The most of them were "Broken pipe" errors:
![brokenpipe](https://user-images.githubusercontent.com/67782524/196175270-4b1a3c36-5cc9-40c6-b104-51bf42038e7a.png)
Do you know how I can fix them ?

**Comment by ljungbaeck** [comment]
Hi guys, Any update on this request? I might be able to send a small reqif-file with date if needed. This would be a nice feature to have since I just got hands of a file containing:
SPEC-TYPES -> ATTRIBUTE-DEFINITION-DATE
DATATYPES -> DATATYPE-DEFINITION-DATE
SPEC-OBJECTS -> ATTRIBUTE-VALUE-DATE

**Reply by stanislaw** [reply]
A small ReqIF file with a date would be super nice to have! Please send it, and I could take care of it within this week.

**Reply by stanislaw** [reply]
The email is on my GitHub profile.

**Comment by stanislaw** [comment, marked as answer]
I have just released the version 0.0.21 which should parse Date and Real without any crashes. If you are still interested in the outcome, please give it a try.

P.S.: @magr2, sorry for not getting back to you about your issues with the dev environment in October this year. I was completely blocked from any open-source work at my day job.

**Reply by stanislaw** [reply]
If you still experience any issues, please send me a minimal reproducible example via email. I will close this discussion for now, feel free to reopen if there is more to implement. Thank you.
